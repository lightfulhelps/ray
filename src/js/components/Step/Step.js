// @flow
import * as React from 'react';
import classNames from 'classnames';
import Icon from '../Icon/Icon';

type Props = {
  activeStep: number,
  isLast: boolean,
  label: string,
  thisStep: number,
  value: number | string,
};

const Step = ({ label, value, isLast, activeStep, thisStep }: Props) => {
  const classes = classNames(
    'stepper__step__value border rounded-circle d-flex align-items-center justify-content-center mr-1',
    // active
    { 'bg-primary text-white border-primary': activeStep === thisStep },
    // completed
    { 'bg-white border-primary': thisStep < activeStep },
    // todo
    { 'bg-gray-400 text-white': thisStep > activeStep }
  );

  const getValue = () => {
    if (thisStep < activeStep) {
      return <Icon name="tick" color="#27b0cc" />;
    }
    return <b>{value}</b>;
  };

  return (
    <div>
      <div className="d-flex align-items-center stepper__step mb-1">
        <div className={classes}>{getValue()}</div>
        <span className="mr-1 text-gray-500">
          <b>{label}</b>
        </span>
        {!isLast && <div className="line d-none d-sm-block mr-1" />}
      </div>
    </div>
  );
};

export default Step;
