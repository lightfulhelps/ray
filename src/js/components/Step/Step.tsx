import * as React from 'react';
import classNames from 'classnames';
import Icon from '../Icon/Icon';

type Props = {
  activeStep: number;
  isLast: boolean;
  label: string;
  onClick?: (event: React.MouseEvent) => void;
  thisStep: number;
  value: number | string;
};

const Step: React.FC<Props> = ({ activeStep, isLast, label, onClick, thisStep, value }: Props) => {
  const isTodo = thisStep > activeStep;
  const isActive = activeStep === thisStep;
  const isCompleted = thisStep < activeStep;

  const stepClasses = classNames('d-flex align-items-center', {
    'mb-1 mb-sm-0 flex-fill': !isLast,
  });

  const valueClasses = classNames(
    'stepper__value border rounded-circle d-flex align-items-center justify-content-center me-1 text-xs fw-bold',
    { 'bg-gray-400 border-gray-400 text-white': isTodo },
    { 'bg-primary text-white border-primary': isActive },
    { 'bg-white border-primary': isCompleted }
  );

  const labelClasses = classNames(
    'me-1',
    { 'text-gray-400 fw-light': isTodo },
    { 'fw-bold text-primary': isActive || isCompleted }
  );

  return (
    <div className={stepClasses}>
      <div
        className={`d-flex align-items-center ${onClick ? 'cursor-pointer' : ''}`}
        data-test-id="stepper-step"
        onClick={onClick}
      >
        <div className={valueClasses}>
          {isCompleted ? <Icon name="tick" theme="primary" /> : value}
        </div>
        <div className={labelClasses}>{label}</div>
      </div>
      {!isLast && <div className="d-none d-sm-block me-1 flex-fill border-top border-gray-400" />}
    </div>
  );
};

export default Step;
