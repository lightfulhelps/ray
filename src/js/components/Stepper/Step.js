// @flow
import * as React from 'react';
import Col from '../Col/Col';
// import stepper from '../../../scss/custom/stepper';

const Step = props => {
  const { label, value, isLast } = props;
  console.log('props', props)
  return (
    <Col>
      <div className="d-flex align-items-center">
        <div className="step-value border border-black rounded-circle d-flex align-items-center justify-content-center mr-1">
          {' '}
          {value}{' '}
        </div>
        <span className="mr-1">{label}</span>
        {!isLast && <div className="line" />}
      </div>
    </Col>
  );
};

export default Step;
