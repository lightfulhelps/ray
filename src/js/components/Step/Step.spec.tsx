import React from "react";
import { shallow } from "enzyme";
import merge from "lodash/merge";
import Step from "./Step";

const setup = (overrides = {}) => {
  const props = merge({
    activeStep: 2,
    isLast: false,
    label: 'step one',
    onClick: jest.fn(),
    thisStep: 2,
    value: 2
  }, overrides);
  const wrapper = shallow(<Step {...props} />);

  return {
    wrapper,
    props
  };
};

describe('<Step />', () => {
  it('should render', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  describe('active state', () => {
    it('should add appropriate classes if the `activeStep` prop is equal to the `thisStep` prop', () => {
      const {
        wrapper
      } = setup();

      expect(wrapper.html()).toContain('bg-primary text-white border-primary');
    });

    it('should render the label prop as the content if the `activeStep` prop is equal to the `thisStep` prop', () => {
      const {
        wrapper,
        props
      } = setup();

      expect(wrapper.find('Icon')).toHaveLength(0);
      expect(wrapper.text()).toContain(props.label);
    });
  });

  describe('todo state', () => {
    it('should add appropriate classes if the `thisStep` prop is greater than the `activeStep` prop', () => {
      const {
        wrapper
      } = setup({ activeStep: 1 });

      expect(wrapper.html()).toContain('bg-gray-400 border-gray-400 text-white');
    });

    it('should render the label prop as the content if the `activeStep` prop is less than the `thisStep` prop', () => {
      const {
        wrapper,
        props
      } = setup();

      expect(wrapper.find('Icon')).toHaveLength(0);
      expect(wrapper.text()).toContain(props.label);
    });
  });

  describe('done state', () => {
    it('should add appropriate classes if the `thisStep` prop is less than the `activeStep` prop', () => {
      const {
        wrapper
      } = setup({ thisStep: 1 });

      expect(wrapper.html()).toContain('bg-white border-primary');
    });

    it('render a tick icon if the `thisStep` prop is less than the `activeStep` prop', () => {
      const {
        wrapper
      } = setup({ thisStep: 1 });

      expect(wrapper.find('Icon')).toHaveLength(1);
    });
  });

  it('should call onClick on click', () => {
    const {
      wrapper,
      props
    } = setup();

    wrapper.find('[data-test-id="stepper-step"]').simulate('click');

    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});