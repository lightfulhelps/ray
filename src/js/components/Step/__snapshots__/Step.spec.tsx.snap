// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> should render 1`] = `
<div
  className="d-flex align-items-center mb-1 mb-sm-0 flex-fill"
>
  <div
    className="d-flex align-items-center cursor-pointer"
    data-test-id="stepper-step"
    onClick={[MockFunction]}
  >
    <div
      className="stepper__value border rounded-circle d-flex align-items-center justify-content-center me-1 text-xs font-weight-bold bg-primary text-white border-primary"
    >
      2
    </div>
    <div
      className="me-1 font-weight-bold text-primary"
    >
      step one
    </div>
  </div>
  <div
    className="d-none d-sm-block me-1 flex-fill border-top border-gray-400"
  />
</div>
`;
